app/controllers/schemes_controller.rb

```diff
@@ -64,23 +72,43 @@ def update
   end
 
   def primary_client_group
-    render "schemes/primary_client_group"
+    if @scheme.confirmed?
```

Comment:
It looks like you're doing the same thing for each of these group actions. Do you think this would help the repetition?

```ruby
before_filter :redirect_if_scheme_confirmed, only: [:primary_client_group, ...]

# ... 

private

def redirect_if_scheme_confirmed
  redirect_to @scheme if @scheme.confirmed?
end
```

Reply:
Awesome! Ty!